tests/integration/security/jwt_test.go

```diff
@@ -409,61 +357,79 @@ func TestIngressRequestAuthentication(t *testing.T) {
+				testCases := []authn.TestCase{
+					{
+						Name: "in-mesh-with-expired-token",
+						CallOpts: echo.CallOptions{
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Headers: map[string][]string{
+								authHeaderKey: {"Bearer " + jwt.TokenExpired},
 							},
-							ExpectResponseCode: response.StatusUnauthorized,
+							Count: callCount,
 						},
-						{
-							Name: "in-mesh-without-token",
-							Request: connection.Checker{
-								From: a[0],
-								Options: echo.CallOptions{
-									Target:   bSet[0],
-									PortName: "http",
-									Scheme:   scheme.HTTP,
-									Count:    callCount,
-								},
-								DestClusters: bSet.Clusters(),
-							},
-							ExpectResponseCode: response.StatusCodeOK,
+						ExpectResponseCode: response.StatusUnauthorized,
+					},
+					{
+						Name: "in-mesh-without-token",
+						CallOpts: echo.CallOptions{
+							PortName: "http",
+							Scheme:   scheme.HTTP,
+							Count:    callCount,
 						},
-					}
-					for _, c := range testCases {
-						t.NewSubTest(c.Name).Run(func(t framework.TestContext) {
-							retry.UntilSuccessOrFail(t, c.CheckAuthn,
-								retry.Delay(250*time.Millisecond), retry.Timeout(30*time.Second))
+						ExpectResponseCode: response.StatusCodeOK,
+					},
+				}
+				for _, c := range testCases {
+					echotest.New(t, apps.All).
+						SetupForPair(func(t framework.TestContext, src, dst echo.Instances) error {
+							policy := yml.MustApplyNamespace(t, tmpl.MustEvaluate(
+								file.AsStringOrFail(t, "testdata/requestauthn/ingress.yaml.tmpl"),
+								map[string]string{
+									"Namespace": ns.Name(),
+									"dst":       dst[0].Config().Service,
+								},
+							), ns.Name())
+							return t.Config().ApplyYAML(ns.Name(), policy)
+						}).
+						From(filters(t, ns.Name(), false)...).
+						ConditionallyTo(echotest.ReachableDestinations).
+						ConditionallyTo(func(from echo.Instance, to echo.Instances) echo.Instances {
+							return to.Match(echo.InCluster(from.Config().Cluster))
+						}).
+						To(filters(t, ns.Name(), false)...).
+						Run(func(t framework.TestContext, src echo.Instance, dest echo.Instances) {
+							t.NewSubTest(c.Name).Run(func(t framework.TestContext) {
+								c.CallOpts.Target = dest[0]
+								c.DestClusters = dest.Clusters()
+								c.CallOpts.Validator = echo.And(echo.ValidatorFunc(c.CheckAuthn))
+								src.CallWithRetryOrFail(t, c.CallOpts, echo.DefaultCallRetryOptions()...)
+							})
 						})
-					}
+				}
+			})
+
+			// TODO(JimmyCYJ): add workload-agnostic test pattern to support ingress gateway tests.
+			policy := yml.MustApplyNamespace(t, tmpl.MustEvaluate(
+				file.AsStringOrFail(t, "testdata/requestauthn/ingress.yaml.tmpl"),
+				map[string]string{
+					"Namespace": ns.Name(),
+					"dst":       util.BSvc,
+				},
+			), ns.Name())
+			t.Config().ApplyYAMLOrFail(t, ns.Name(), policy)
+			t.NewSubTest("").Run(func(t framework.TestContext) {
```

Comment:
Could you give this sub test a name to distinguish from the one in line 368?

Reply:
Done.